services/src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -223,7 +236,11 @@ public double measure(MetricConfig config, long now) {
             double availabilitySum = 0.0;
             int partitionNum = _partitionNum.get();
             for (int partition = 0; partition < partitionNum; partition++) {
-              double recordsProduced = _sensors.metrics.metrics().get(new MetricName("records-produced-rate-partition-" + partition, METRIC_GROUP_NAME, tags)).value();
+              MetricName partitionRecordsProduced = new MetricName("records-produced-rate-partition-" + partition, METRIC_GROUP_NAME, tags);
+              if (!_sensors.metrics.metrics().containsKey(partitionRecordsProduced)) {
```

Comment:
I am not sure that we need this check. `addPartitionSensors(...)` should have created sensors for all partitions, right?